src/uproot/models/RNTuple.py

```diff
@@ -480,10 +493,27 @@ def read_pagedesc(self, destination, desc, dtype_str, dtype, nbits, split):
         # needed to chop off extra bits incase we used `unpackbits`
         destination[:] = content[:num_elements]
 
-    def read_col_pages(self, ncol, cluster_range, pad_missing_ele=False):
-        res = numpy.concatenate(
-            [self.read_col_page(ncol, i) for i in cluster_range], axis=0
-        )
+    def read_col_pages(self, ncol, cluster_range, dtype_byte, pad_missing_ele=False):
+        arrays = [self.read_col_page(ncol, i) for i in cluster_range]
+
+        # Check if column stores offset values for jagged arrays (splitindex64) (applies to cardinality cols too):
+        if (
+            dtype_byte == uproot.const.rntuple_col_type_to_num_dict["splitindex64"]
+            or dtype_byte == uproot.const.rntuple_col_type_to_num_dict["splitindex32"]
+        ):
+            # Continue new cluster values from the last value of previous cluster.
+
+            # Extract the last offset values:
+            last_elements = [arr[-1] for arr in arrays[:-1]]
+            # Compute cumulative sum using itertools.accumulate:
+            last_offsets = [0, *list(accumulate(last_elements))]
+            # Add the offsets to each array
+            arrays = [arr + offset for arr, offset in zip(arrays, last_offsets)]
```

Comment:
It would be better to try to do operations in place if possible.
```suggestion
            for arr, offset in zip(arrays, last_offsets):
                arr += offset
```

Also, you could skip the first one since the offset is always 0

Reply:
I see, indeed, your suggestion helps to save used memory. 

I have just committed both of these suggestions.